worldedit-core/src/main/java/com/sk89q/worldedit/command/GeneralCommands.java

```diff
@@ -129,25 +133,67 @@ public void timeout(Actor actor, LocalSession session,
-            return;
+            @Arg(desc = "The side effect", def = "")
+                    SideEffect sideEffect,
+            @Arg(desc = "The new side effect state", def = "")
+                    SideEffect.State newState,
+            @Switch(name = 'h', desc = "Show the info box")
+                    boolean showInfoBox) {
+        if (sideEffect != null) {
+            SideEffect.State currentState = session.getSideEffectSet().getState(sideEffect);
+            if (newState != null && newState == currentState) {
+                if (!showInfoBox) {
+                    actor.printError(TranslatableComponent.of(
+                            "worldedit.fast.sideeffect.already-set",
+                            TranslatableComponent.of(sideEffect.getDisplayName()),
+                            TranslatableComponent.of(newState.getDisplayName())
+                    ));
+                }
+                return;
+            }
+
+            if (newState != null) {
+                session.setSideEffectSet(session.getSideEffectSet().with(sideEffect, newState));
+                if (!showInfoBox) {
+                    actor.printInfo(TranslatableComponent.of(
+                            "worldedit.fast.sideeffect.set",
+                            TranslatableComponent.of(sideEffect.getDisplayName()),
+                            TranslatableComponent.of(newState.getDisplayName())
+                    ));
+                }
+            } else {
+                actor.printInfo(TranslatableComponent.of(
+                        "worldedit.fast.sideeffect.get",
+                        TranslatableComponent.of(sideEffect.getDisplayName()),
+                        TranslatableComponent.of(currentState.getDisplayName())
+                ));
+            }
+        } else if (newState != null) {
+            SideEffectSet applier = session.getSideEffectSet();
+            for (SideEffect sideEffectEntry : SideEffect.values()) {
+                applier = applier.with(sideEffectEntry, newState);
+            }
+            session.setSideEffectSet(applier);
+            if (!showInfoBox) {
+                actor.printInfo(TranslatableComponent.of(
+                        "worldedit.fast.sideeffect.set-all",
+                        TranslatableComponent.of(newState.getDisplayName())
+                ));
+            }
         }
 
-        if (hasFastMode) {
-            session.setFastMode(false);
-            actor.printInfo(TranslatableComponent.of("worldedit.fast.disabled"));
-        } else {
-            session.setFastMode(true);
-            actor.printInfo(TranslatableComponent.of("worldedit.fast.enabled"));
+        if (sideEffect == null || showInfoBox) {
+            SideEffectBox sideEffectBox = new SideEffectBox(session.getSideEffectSet());
+            try {
+                actor.print(sideEffectBox.create(1));
+            } catch (InvalidComponentException e) {
+                e.printStackTrace();
```

Comment:
re-throw wrapped as CommandException? shouldn't happen but if it does the user gets no feedback in chat

Reply:
I copied this from WorldGuard code :P

Reply:
ouch, got me.